src/interrupts/src/lib.rs

```diff
@@ -168,32 +173,24 @@ pub extern "C" fn interrupt_handler(interrupt_number: isize, error_code: isize)
         33 => keyboard_handler(),
         _ => panic!("interrupt {} with error code {:x}", interrupt_number, error_code),
     }
-    unsafe{
-        match interrupt_number {
-            i if i >= 40 => {
-                outb(0xA0, 0x20);
-                outb(0x20, 0x20);
-            },
-            32...40 => outb(0x20, 0x20),
-            _ => {},
-        }
-        enable();
-    };
+
+    pic::eoi_for(interrupt_number);
+
+    enable();
 }
 
 fn keyboard_handler() {
     let scancode = unsafe { inb(0x60) };
-    kprintln!("code: {}", scancode);
+
+    keyboard::getchar(scancode).map(|c| {
+        keyboard::write_to_buffer(c);
+        kprint!("{}", c as char);
+    });
 }
 
 #[inline]
-pub unsafe fn inb(port: u16) -> u8 {
+unsafe fn inb(port: u16) -> u8 {
```

Comment:
Yes. I am planning on porting all of it to that crate, 👍

So I just left the duplication in for now.


On Jun 12, 2016, 16:23 +0300, nxnfufunezn<notifications@github.com>, wrote:
>  
> Insrc/interrupts/src/lib.rs(https://github.com/intermezzOS/kernel/pull/40#discussion_r66720728):
>  
> >}>>#[inline]>-pub unsafe fn inb(port: u16) ->u8 {>+unsafe fn inb(port: u16) ->u8 {
>  
> same function defined at src/pic/src/lib.rs..
> this crate can also be usedhttps://github.com/gz/rust-x86..
>  
>  
> —
> You are receiving this because you authored the thread.
> Reply to this email directly,view it on GitHub(https://github.com/intermezzOS/kernel/pull/40/files/dcf3fe3f039eea9de157c6a489dad21213adfd11#r66720728), ormute the thread(https://github.com/notifications/unsubscribe/AABsirmVbPg8OGqEU0YJjUSEoJFegJlgks5qLAhhgaJpZM4Iy3r-).
>  
>  
>